### PR TITLE
feat(book): allow F* extraction + TC from within the book

### PR DESCRIPTION
**Pull request by W95Psp**
This PR:
 - fixes the book (things were broken because we dropped the theme, that contained custom code I wrote)
 - add buttons to extract and typecheck from within the book, using the API of the playground
 - do some refactor, add a introduction for the examples

### CODE REVIEW

File: book/src/examples/intro.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Examples
+
+This chapter contains various examples that demonstrates how hax can
+be used to prove properties about programs. Each example is
+self-contained. hax being a tool that can extract Rust to various
+backends, this section provides examples for each backends..
```
Comment (maximebuyse) [inline book/src/examples/intro.md:6]: "each backend" (without s), and probably only one dot at the end.

**Review by maximebuyse — CHANGES_REQUESTED** [review]
Just noticed a typo but it looks great otherwise!

File: book/src/examples/intro.md (hunk 1 above)
Comment (W95Psp) [reply book/src/examples/intro.md:6]: Good catch, thanks!

**Review by maximebuyse — APPROVED** [review]

**Review by karthikbhargavan — APPROVED** [review]
This needs some text polish, but let's merge this PR so that the code/checking features are merged.

**Pull request merged**